<?xml version="1.0"?>
<launch>

  <arg name="no_static_map" default="false"/>
  <arg name="base_global_planner" default="navfn/NavfnROS"/>
<!--
  <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>
-->
  <arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS"/>

<!--
  <arg name="node_name"             value="velocity_smoother"/>
  <arg name="nodelet_manager_name"  value="nodelet_manager"/>
  <arg name="config_file"           value="$(find yocs_velocity_smoother)/param/standalone.yaml"/>
  <arg name="raw_cmd_vel_topic"     value="cmd_vel"/>
  <arg name="smooth_cmd_vel_topic"  value="komodo_1/cmd_vel"/>
  <arg name="robot_cmd_vel_topic"   value="robot_cmd_vel"/>
  <arg name="odom_topic"            value="komodo_1/odom_pub"/>
-->

<!-- cancel the master auto start  

<master auto="start"/>
-->  
  


 <!-- Run the map server 

  <node name="map_server" pkg="map_server" type="map_server" args="$(find webot_navigation)/maps/map.yaml"/> 

-->

<!-- nodelet manager 
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager"/> 
-->
<!-- velocity smoother 
 <include file="$(find yocs_velocity_smoother)/launch/velocity_smoother.launch">
    <arg name="node_name"             value="$(arg node_name)"/>
    <arg name="nodelet_manager_name"  value="$(arg nodelet_manager_name)"/>
    <arg name="config_file"           value="$(arg config_file)"/>
    <arg name="raw_cmd_vel_topic"     value="$(arg raw_cmd_vel_topic)"/>
    <arg name="smooth_cmd_vel_topic"  value="$(arg smooth_cmd_vel_topic)"/>
    <arg name="robot_cmd_vel_topic"   value="$(arg robot_cmd_vel_topic)"/>
    <arg name="odom_topic"            value="$(arg odom_topic)"/>
  </include>
-->



  <!-- use remap to try to fix /odom 
   <remap from="/odom" to="/komodo_1/odom_pub"/>  --> 
  
  <!--- Run AMCL 
   <include file="$(find webot_navigation)/launch/ric_amcl.launch" /> -->   

  <!--- Run rviz --> 
   <node name="rviz" pkg="rviz" type="rviz" args="-d $(find webot)/rviz/robot.rviz" output="screen"/>

  <!---  do not use this 
  <include file="$(find webot_navigation)/launch/gmapping_slam.launch">
       <arg name="base_frame" value="base_link"/>
       <arg name="odom_frame" value="odom"/>
  </include>  --> 

   <!--- Run gmapping 
  <include file="$(find webot_navigation)/launch/gmapping.launch" /> -->

  <!--- Run hector_slam 
  <include file="$(find webot_navigation)/launch/hector_slam.launch"> 
       <arg name="base_frame" value="base_link"/>
       <arg name="odom_frame" value="odom"/>
       <param name="pub_map_odom_transform" value="false"/>
  </include> -->

  <!--- Run move_base --> 
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/> 
    <param name="controller_frequency" value="20"/>
    <param name="planner_frequency" value="20"/>
    <!--- <remap from="cmd_vel" to="input/nav_cmd_velocity"/> -->
    <rosparam file="$(find webot_navigation)/config/planner.yaml" command="load"/>

    <!-- observation sources located in costmap_common.yaml -->
    <rosparam file="$(find webot_navigation)/config/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find webot_navigation)/config/costmap_common.yaml" command="load" ns="local_costmap" />

    <!-- local costmap, needs size -->
    <rosparam file="$(find webot_navigation)/config/costmap_local.yaml" command="load" ns="local_costmap" />
    <param name="local_costmap/width" value="10.0"/>
    <param name="local_costmap/height" value="10.0"/>

    <!-- static global costmap, static map provides size -->
    <rosparam file="$(find webot_navigation)/config/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/>
    
    <!-- global costmap with laser, for odom_navigation_demo -->
    <rosparam file="$(find webot_navigation)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)"/>
    <param name="global_costmap/width" value="100.0" if="$(arg no_static_map)"/>
    <param name="global_costmap/height" value="100.0" if="$(arg no_static_map)"/>

    <!--<remap from="odom" to="/komodo_1/odom_pub"/> -->
  </node>

</launch>
